### PR TITLE
fix issue regarding item synchronisation

### PR DESCRIPTION
**Pull request by x4d3**

### CODE REVIEW

**Comment by x4d3** [comment]
@Berardpi Could you please review ? Thanks,

File: app/models/entities/item.rb [hunk 1]
```diff
@@ -28,106 +36,143 @@ def consolidate_and_map_data(connec_entities, external_entities, organization, o
...
     mapped_external_entities_with_idmaps.each do |mapped_external_entities_with_idmap|
       parent_connect_id = mapped_external_entities_with_idmap[:idmap].connec_id
       product = mapped_external_entities_with_idmap[:entity]
       product[:variants].each do |variant|
         variant[:parent_item_id] = parent_connect_id
-        idmap = Maestrano::Connector::Rails::IdMap.find_by(external_id: variant[:external_id], connec_entity: connec_entity_name.downcase, external_entity: 'variant', organization_id: organization.id)
-        variants.push({entity: variant, idmap: idmap || create_id_map(variant, organization)})
+        idmap = Maestrano::Connector::Rails::IdMap.find_by(external_id: variant[:external_id], connec_entity: self.class.connec_entity_name.downcase, external_entity: 'variant', organization_id: organization.id)
+        variants.push({entity: variant, idmap: idmap || create_variant_id_map(variant, organization)})
       end if product[:variants]
```
Comment (Berardpi) [inline app/models/entities/item.rb]: not so fan of the if here
would rather have put a `next unless product[:variants]` before the loop

File: app/models/entities/item.rb [hunk 2]
```diff
@@ -28,106 +36,143 @@ def consolidate_and_map_data(connec_entities, external_entities, organization, o
...
+          variant[:id] = idmap.external_id if idmap
+          variant[:product_id] = product_id
+        end
       end
+
     end
     super(external_client, mapped_connec_entities_with_idmaps, organization)
   end
 
   private
```
Comment (Berardpi) [inline app/models/entities/item.rb]: I won't comment about the indentation. Oups, too late.

**Comment by Berardpi** [comment]
Looks good! Will do some testing to break it once released :smiling_imp: 

(just your ide changing the indentation is messing up with the diff :crying_cat_face:)

**Pull request merged**